packages/react/src/checkbox/Checkbox.css.ts

```diff
@@ -0,0 +1,125 @@
+import { theme } from "../styles";
+import { recipe, style } from "../vanilla-extract";
+
+export const wrapper = recipe({
+  base: [
+    {
+      alignItems: "start",
+      flexDirection: "row",
+      gap: "xs",
+    },
+  ],
+});
+
+export const iconC = recipe({
+  base: [
+    style({
+      display: "none",
+      selectors: {
+        '[data-state="checked"] &': {
```

Comment:
we shouldn't use unscoped css here - like we've done for `[data-state="checked"]`

let's add a class selector to it like this:

```ts
const wrapperMarker = style({});

export const wrapper = recipe({
  base: [
    {
      alignItems: "start",
      flexDirection: "row",
      gap: "xs",
    },
   wrapperMarker,
  ],
});

export const iconC = recipe({
  base: [
    style({
      display: "none",
      selectors: {
        [`${wrapperMarker}[data-state="checked"] &`]: {
          display: "block",
        },
      },
    }),
  ],
});
```

Reply:
we might not have issues with `[data-state="checked"]` - but for something like `[data-disabled="true"]` an ancestor element higher in the tree might have that attribute for whatever reason and will trigger this selector unintentionally

whereas if we use a marker like this then we're guaranteed to only look at attributes on the marker

Reply:
interesting bhai